### PR TITLE
Dennis/feed to profile navigation

### PR DESCRIPTION
**Pull request by dennisfedorko**
Clicking on Users Picture in Feed Jumps to Their Profile Page #46

### CODE REVIEW

File: IceFishing/Views/PostView.swift [hunk 1]
```diff
@@ -292,9 +292,12 @@ class PostView: UIView, UIGestureRecognizerDelegate {
 			likedButton?.setImage(UIImage(named: name), forState: .Normal)
 		} else if hitView == addButton {
 			print("Adding")
-		} else if post.player.isPlaying() {
+		} else {
 			if hitView == avatarImageView || hitView == self.profileNameLabel {
 				// GO TO PROFILE VIEW CONTROLLER
+				if let cell = superview?.superview as? FeedTableViewCell {
```
Comment (lucasderraugh) [inline IceFishing/Views/PostView.swift]: We should make this a more generic `-cell:didTapImage:` delegate. Could be used across the board for when images are selected. Using `superview` property is dangerous in the case where the view hierarchy changes.

**Comment by lucasderraugh** [comment]
Look good, but we shouldn't directly reference the parenting controller through superviews. Make the connection more direct between the 2 elements.

**Comment by lucasderraugh** [comment]
Add in to the PostViewDelegate (in PostView) for your call to move to the profile.

**Comment by dennisfedorko** [comment]
I also fixed the navigation so it shows "Back" instead of a hamburger menu if you navigate from feed to profile to some other controller

File: IceFishing/Views/PostView.swift [hunk 2]
```diff
@@ -9,8 +9,9 @@
 import UIKit
 import MediaPlayer
 
-protocol PostViewDelegate {
+@objc protocol PostViewDelegate {
 	func didTapAddButtonForCell()
+	optional func showUserProfile(user:User)
```
Comment (lucasderraugh) [inline IceFishing/Views/PostView.swift]: Make this `didTapUserProfile` as a delegate method.

File: IceFishing/Controllers/FeedViewController.swift [hunk 3]
```diff
@@ -338,4 +338,15 @@ class FeedViewController: UITableViewController, SongSearchDelegate, PostViewDel
 		}
 
 	}
+	
+	//MARK: -
+	//MARK: Navigation
+	
+	func showUserProfile(user:User) {
```
Comment (lucasderraugh) [inline IceFishing/Controllers/FeedViewController.swift]: ...from my other comment, this method name should reflect the delegate method.

File: IceFishing/Profile/UsersViewController.swift [hunk 4]
```diff
@@ -37,11 +40,15 @@ class UsersViewController: UITableViewController {
...
-        navigationItem.leftBarButtonItem = UIBarButtonItem(customView: backButton)
+		//Navigation
+		if pushNavigationEnabled {
+			// Add back button to profile
+			let backButton = UIButton(frame: CGRect(x: 0, y: 0, width: 50, height: navigationController!.navigationBar.frame.height))
+			backButton.setImage(UIImage(named: "Close-Icon"), forState: .Normal)
+			backButton.setTitle("Back", forState: UIControlState.Normal)
+			backButton.addTarget(self, action: "popToPrevious", forControlEvents: .TouchUpInside)
+			navigationItem.leftBarButtonItem = UIBarButtonItem(customView: backButton)
+		}
```
Comment (lucasderraugh) [inline IceFishing/Profile/UsersViewController.swift]: This shouldn't be necessary after I push the change to always show the back button after pushing to the UsersViewController.

File: IceFishing/Views/PostView.swift [hunk 5]
```diff
@@ -9,8 +9,9 @@
 import UIKit
 import MediaPlayer
 
-protocol PostViewDelegate {
+@objc protocol PostViewDelegate {
 	func didTapAddButtonForCell()
```
Comment (lucasderraugh) [inline IceFishing/Views/PostView.swift]: If you have the chance, can you also make this method optional. This entire delegate should have optional methods.

**Comment by lucasderraugh** [comment]
Close #46

**Pull request closed without merging**